### ISSUE TITLE
Poor inference results on own dataset

### ISSUE THREAD

**Issue by gaishi7** [liuqk3/PUT#40]
![0b67ab49e96604b3e99fc949991133e](https://github.com/liuqk3/PUT/assets/74124390/439d6631-a9cf-48d0-a70a-5761cde7da00)
![4369556c30c7ec5a744e15abe324004](https://github.com/liuqk3/PUT/assets/74124390/2b623a30-6316-4227-abf9-73c6420a1430)
![db3289873aea61e95f9cf672bd667db](https://github.com/liuqk3/PUT/assets/74124390/82fdd060-cc46-45ba-b55b-22be53ba23d8)
Hello, when training on my own dataset and using the command python scripts/inference.py --func inference_inpainting --name OUTPUT/cvpr2022_transformer_ffhq/checkpoint/last.pth --input_res 256,256 --num_token_per_iter 100 --num_token_for_sampling 300 --num_replicate 1 --image_dir data/1 --mask_dir irregular-mask/2 --save_masked_image --save_dir out_images/cvpr2022_transformer_ffhq --num_sample 1 --gpu 0, the inference results are very poor. How can this be resolved?

**Comment by liuqk3** [comment]
Thanks for your interest in our work. You need to make sure that
(1)  the P-VQVAE can well reconstruct your own images. From my experience, P-VQVAE may fails to reconstruct the textures in your provided example images;
(2) the transformer can generate pleasant results on your training data. If the P-VQVAE fails to reconstruct the images well, it is hard for the transformer to produce pleasant inpainted images.

**Comment by gaishi7** [comment]
Thank you for your response.Is it possible to solve this problem by tuning the parameters?

**Comment by gaishi7** [comment]
Thank you for your response.This is a real image.
![cdc24ddb454f801166f00e080a2c493](https://github.com/user-attachments/assets/03650d54-0cab-4d1a-82c9-e4509be6f646)
This is a reconstructed image.
![5ec69be271977f8182fdb654e48770f](https://github.com/user-attachments/assets/056fc40d-6ede-459b-a69d-552c35548a8d)
This is a restored image.
![add82eba34da72f06aa1af5f7f068dd](https://github.com/user-attachments/assets/416f0f5f-1225-4fd9-b202-0df549698fa3)
Why does the reconstruction image look great, but the restoration result is poor？

**Comment by gaishi7** [comment]
> Thank you for your response.This is a real image. ![cdc24ddb454f801166f00e080a2c493](https://private-user-images.githubusercontent.com/74124390/363067910-03650d54-0cab-4d1a-82c9-e4509be6f646.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTAxNTAsIm5iZiI6MTcyNjEwOTg1MCwicGF0aCI6Ii83NDEyNDM5MC8zNjMwNjc5MTAtMDM2NTBkNTQtMGNhYi00ZDFhLTgyYzktZTQ1MDliZTZmNjQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDAyNTczMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzYjdmMWI4NDQwNzZmMjIwOWYzZGJkNjhmYmI3MTBkNDIwMjlkZGI4YTA4MjY2ZDk4ODEwN2FkNmFhOWZjMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HIuZgBhUrJDmWkxsyLxRwWeUD7yavJ7FvfvP91jlmwc) This is a reconstructed image. ![5ec69be271977f8182fdb654e48770f](https://private-user-images.githubusercontent.com/74124390/363068449-056fc40d-6ede-459b-a69d-552c35548a8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTAxNTAsIm5iZiI6MTcyNjEwOTg1MCwicGF0aCI6Ii83NDEyNDM5MC8zNjMwNjg0NDktMDU2ZmM0MGQtNmVkZS00NTliLWE2OWQtNTUyYzM1NTQ4YThkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDAyNTczMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzODFiZjBlNTdmZjViM2IzNmFhYTQ1NDBjZmQxMzFmOTUwYzk5OWRiNDhhMDUxN2ZmYWIxNTFiYzFjMDcyYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ombjiVxsO-p2oKBPoV_EmulBU9OZLAKuZ3rQRFs3esk) This is a restored image. ![add82eba34da72f06aa1af5f7f068dd](https://private-user-images.githubusercontent.com/74124390/363068650-416f0f5f-1225-4fd9-b202-0df549698fa3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxMTAxNTAsIm5iZiI6MTcyNjEwOTg1MCwicGF0aCI6Ii83NDEyNDM5MC8zNjMwNjg2NTAtNDE2ZjBmNWYtMTIyNS00ZmQ5LWIyMDItMGRmNTQ5Njk4ZmEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDAyNTczMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZTNhYjg4NmMzZmNmM2Y4MmJlNTlhMWU1NGY0OWMxZjlkZjMzNDFkOWNiMGY1MDQ0MWM5ZWQ1OWZkN2VhZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZJ59ZYW1wa0XBllQ1-ohgXToleBY6EMHFe6AJ6Imyac) Why does the reconstruction image look great, but the restoration result is poor？
Could you please reply if you see this?

**Comment by liuqk3** [comment]
Which model do you use for this restoration? From my experience, the unconverged model (UQ-Transformer) will produce similar results with yours restored image. And the image you provided is quite different from the three datasets I used. You may fine-tune the model on the own dataset for better performance.
